Сергеева Наталья

Pogromistka opened the pull request.

Pogromistka committed:
Все на вечеринку

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
исправлены ошибки

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
ошибки поправлены

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
Вечеринка

honest-hrundel commented:
:tomato: Пройдено тестов 9 из 18

Pogromistka committed:
Вариант 2

honest-hrundel commented:
:tomato: Пройдено тестов 9 из 18

Pogromistka committed:
Без console

honest-hrundel commented:
:tomato: Пройдено тестов 9 из 18

Pogromistka committed:
Нельзя менять collection

honest-hrundel commented:
:tomato: Пройдено тестов 12 из 18

Pogromistka committed:
and or в приоритетах

honest-hrundel commented:
:tomato: Пройдено тестов 12 из 18

Pogromistka committed:
просто для проверки

honest-hrundel commented:
:tomato: Пройдено тестов 10 из 15

Pogromistka committed:
сортировка?

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
сортировка в2

honest-hrundel commented:
:green_apple: Пройдено тестов 15 из 15

Pogromistka committed:
Доп задание

honest-hrundel commented:
:tomato: Пройдено тестов 17 из 18

Pogromistka committed:
17/18

honest-hrundel commented:
:tomato: Пройдено тестов 17 из 18

Pogromistka committed:
fix or

honest-hrundel commented:
:tomato: Пройдено тестов 17 из 18

Pogromistka committed:
fix and

honest-hrundel commented:
:tomato: Пройдено тестов 17 из 18

Pogromistka committed:
func

honest-hrundel commented:
:tomato: Пройдено тестов 17 из 18

Pogromistka committed:
не портить оригиналы?

honest-hrundel commented:
:tomato: Пройдено тестов 17 из 18

Pogromistka committed:
fix or v2

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
убрала неиспользуемую переменную

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
исправила or v3

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
невнимательность..

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
в4

honest-hrundel commented:
:tomato: Пройдено тестов 17 из 18

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,159 @@
  */
 exports.isStar = true;
 
+var assort = ['and', 'or', 'filterIn', 'sortBy', 'select', 'limit', 'format'];
```
Мне кажется, что не очень удачное наименование переменной

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,159 @@
+    var arg = [].slice.call(arguments);
+
+    arg.splice(0, 1);
+    arg.sort(function (argA, argB) {
```
Можно вынести сортировку в отдельную функцию

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,159 @@
+        return assort.indexOf(argA.func) - assort.indexOf(argB.func);
+    });
+
+    for (var i = 0; i < arg.length; i++) {
```
Не хочешь попробовать через forEach?

Pogromistka replied:
Хочу) Во втором исправлении сейчас попробую.. Мне поменять во всех местах, где с массивом работаю?

Pogromistka replied:
ааа, если бы не Вы, за новые для меня методы работы с массивами вообще не взялась бы)) и тесты прошла полностью (or переписала). ураа) спасибо огроменное))
жду ответов)

onufrienko replied:
Да не за что. Но filter, reduce, map.. вообще крутая вещь :)

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,159 @@
+    });
+
+    for (var i = 0; i < arg.length; i++) {
+        copyCollection = arg[i].res(copyCollection);
```
Вот если не смотреть другие методы и не знать заранее  о том, что здесь происходит, а смотреть только на метод query, то не понятно, чем здесь является res

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,159 @@
+
+    return {
+        func: 'select',
+        res: function (collection) {
```
может, назвать поля иначе? например, name и func

Pogromistka replied:
спасибо, сейчас постараюсь всё поправить

onufrienko commented:
Я досмотрю твой пр вечером :)

Pogromistka committed:
Исправления

Поправила неудачно выбранные имена переменных. Вынесла функцию сортировки sortPriorityFunc

honest-hrundel commented:
:tomato: Пройдено тестов 17 из 18

Pogromistka committed:
Исправление 2

forEach

honest-hrundel commented:
:tomato: Пройдено тестов 17 из 18

Pogromistka committed:
17/18

honest-hrundel commented:
:green_apple: Пройдено тестов 18 из 18

honest-hrundel commented:
@onufrienko обрати внимание решено доп. задание

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,165 @@
 };
 
+function copyNoteCollection(note, fields) {
+
```
Лишняя строка, как и в других функциях.

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,165 @@
+    return {
+        name: 'limit',
+        func: function (collection) {
+            if (collection.length > count) {
```
почему не просто splice(0, count)?

Pogromistka replied:
эм.. пыталась так, но что-то не заладилось. Сейчас поменяю

Pogromistka replied:
аа, не запомнила, что возвращает splice. Возвращает массив из удалённых элементов. (источник [learn.javascript](https://learn.javascript.ru/array-methods))

onufrienko commented:
я правильно понимаю, что ты возвращаешь объект с двумя полями - именем функции и самой функцией, чтобы сортировать функции по приоритету перед их выполнением?
если да, то можно же попробовать сделать проще

Pogromistka commented:
да, всё правильно понимаете) другого варианта не придумалось, чтоб узнать имя функции. думала возвращать числа как приоритет и функцию, но потом вдруг что-то поменяется, по коду бегать исправлять не очень.. а как сделать проще?

onufrienko commented:
Давай на ты :)
Мне сейчас снова надо убегать, надеюсь, вечером отвечу. Ну или утром завтра

Pogromistka committed:
Исправление по замечаниям

splice и лишние строки

honest-hrundel commented:
:green_apple: Пройдено тестов 18 из 18

honest-hrundel commented:
@onufrienko обрати внимание решено доп. задание

onufrienko commented:
Что если переписать функции через классные методы JS? :)

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,158 @@
+        func: function (collection) {
 
-    return;
+            return collection.splice(0, count);
```
давай все же делать slice, чтобы не обрезать коллекцию?

Pogromistka replied:
поправила. из классных методов применился только forEach

Pogromistka committed:
Проба

slice
в select на forEach перешла

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
Проба

slice
и в select перешла на forEach

honest-hrundel commented:
:green_apple: Пройдено тестов 18 из 18

honest-hrundel commented:
@onufrienko обрати внимание решено доп. задание

Pogromistka committed:
на forEach

filterIn, format

honest-hrundel commented:
:green_apple: Пройдено тестов 18 из 18

honest-hrundel commented:
@onufrienko обрати внимание решено доп. задание

Pogromistka committed:
окончательный вариант, наверное

в and forEach прописала)

honest-hrundel commented:
:green_apple: Пройдено тестов 18 из 18

honest-hrundel commented:
@onufrienko обрати внимание решено доп. задание

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,172 @@
  */
 exports.query = function (collection) {
-    return collection;
+    var copyCollection = JSON.parse(JSON.stringify(collection));
```
В одном из пр было сделано вот такое замечание:
![image](https://cloud.githubusercontent.com/assets/14929766/20101822/686bdec0-a5e5-11e6-8cb2-e690423e2a5e.png)
Кидаю его тебе, чтобы ты тоже знала :)

bratva replied:
Нужно избавиться от этого способа клонировать объект, можно учесть что это простой-плоский объект

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,172 @@
+    arg.splice(0, 1);
+    arg.sort(sortPriorityFunc);
+
+    // for (var i = 0; i < arg.length; i++) {
```
может, лучше убрать  комментарии?

Pogromistka replied:
хорошо, я всё, что смогу, уберу и поправлю) спасибочки большое

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,172 @@
+            collection.forEach(function (item) {
+                selectedCol.push(copyNoteCollection(item, fields));
+            }
+            );
```
не нужно переносить скобочку на отдельную строку

bratva replied:
отступы

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,172 @@
+function copyNoteCollection(note, fields) {
+    var remarkCol = {};
+
+    for (var key in note) {
```
Object.keys?

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,172 @@
+                    filteredCol.push(item);
+                }
+            }
+            );
```
тут тоже круглую скобку обратно к фигурной

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,172 @@
+            //    }
+            // }
+            collection.forEach(function (item) {
+                if (item[property] !== undefined && values.indexOf(item[property]) !== -1) {
```
можно просто попробовать `if (item[property]  && values.indexOf(item[property]) !== -1)`, так как если будет   равно undefined, то вернется false

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,172 @@
+                   // return personaB[property] - personaA[property];
+                // }
+
+                return (order === 'asc') ? personaA[property] > personaB[property]
```
firstPerson, secondPerson?

onufrienko replied:
или first/second friend

Pogromistka replied:
аа) ну да, так понятней будет

onufrienko commented on lego.js:
```diff
@@ -6,65 +6,172 @@
+            });
 
-    return;
+            return collection;
```
выше ты возвращаешь filteredCol, remarkCol, а здесь просто collection. Давай тут так же, как и в других методах назовем для единообразия

onufrienko replied:
И в методе ниже тоже

Pogromistka replied:
Дак там были же объявлены новые переменные объекты. А тут получается сортирую и возвращаю переданную collection, т.е. не вижу смысла копировать его. В общем не совсем понимаю тебя.  :-(

onufrienko commented:
Ты умничка, что переписала по-другому :) 
Выглядит лучше, еще немного исправь, а потом уже ментор будет смотреть

onufrienko commented:
🚀

bratva commented on lego.js:
```diff
@@ -6,65 +6,172 @@
 };
 
+function sortPriorityFunc(argA, argB) {
+
```
лишняя строка

bratva commented on lego.js:
```diff
@@ -6,65 +6,172 @@
+            //        filteredCol.push(collection[i]);
+            //    }
+            // }
+            collection.forEach(function (item) {
```
посмотри на метод `filter`

bratva commented on lego.js:
```diff
@@ -6,65 +6,172 @@
+        name: 'sortBy',
+        func: function (collection) {
+            collection.sort(function (personaA, personaB) {
+                // if (order === 'desc') {
```
зачем это здесь?

Pogromistka replied:
не понимаю вопроса. Вы про комментарии? Их убрала

bratva commented on lego.js:
```diff
@@ -73,17 +180,69 @@ if (exports.isStar) {
+                // for (var i = 0; i < arg.length; i++) {
+                //    andCollection = arg[i].func(andCollection);
+                // }
+                arg.forEach(function (item) {
```
Посмотри на метод `every`

bratva commented on lego.js:
```diff
@@ -73,17 +180,69 @@ if (exports.isStar) {
+            func: function (collection) {
+                var andCollection = collection.slice();
+                // for (var i = 0; i < arg.length; i++) {
+                //    andCollection = arg[i].func(andCollection);
```
эти комментарии тоже лишние

bratva commented:
🍎 Нужно исправить все замечания

Pogromistka committed:
исправлены маленькие недочеты)

удалены комментарии и лишние переносы
на object.keys перешла.

honest-hrundel commented:
:green_apple: Пройдено тестов 18 из 18

honest-hrundel commented:
@bratva обрати внимание решено доп. задание

Pogromistka committed:
Избавление от JSON

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
Клонирование collection

honest-hrundel commented:
:tomato: Пройдено тестов 15 из 18

Pogromistka committed:
Клонирование collection v2

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
Строка перед return

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
клонирование collection v3

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

Pogromistka committed:
клонирование collection v4

honest-hrundel commented:
:green_apple: Пройдено тестов 18 из 18

honest-hrundel commented:
@bratva обрати внимание решено доп. задание

bratva commented on lego.js:
```diff
@@ -73,17 +162,48 @@ if (exports.isStar) {
+            name: 'and',
+            func: function (collection) {
+                var andCollection = collection.slice();
+                arg.forEach(function (item) {
```
Попробуй использовать https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every
